pam_nologin: explicitly fail when nologin path is a directory

pinotree opened the pull request:
On some systems (e.g. GNU/Hurd), `read()` succeeds on the fd of a directory; since the module assumes that `read()` fails (and thus `pam_modutil_read()` as well), manually fail in case the open fd refers to a directory.

ldv-alt reviewed (commented):
After this change, a `fstat` error is treated as if there is no file.
I don't mind, but maybe add a comment or mention it somehow?

pinotree commented:
> After this change, a `fstat` error is treated as if there is no file. I don't mind, but maybe add a comment or mention it somehow?

ah I see what you mean: I moved the `fstat` block up as optimization, and that can give a different `retval`; moved it back to where it was, thanks!

ldv-alt reviewed (commented):
Looks good.

As the rest of the file is indented with tabs, could you reindent the new code with tabs, please?

pinotree commented:
> As the rest of the file is indented with tabs, could you reindent the new code with tabs, please?

Oops, done, thanks for noticing. The mixup of tabs and spaces to indent around the sources makes things confusing...
